lib/pagseguro/subscription.rb

```diff
@@ -44,6 +65,56 @@ def errors
       @errors ||= Errors.new
     end
 
+    # Find subscription by notification code
+    def self.find_by_notification_code(code, options={})
+      load_from_response send_request("pre-approvals/notifications/#{code}", options[:credentials])
+    end
+
+    # Find subscription by subscription code
+    def self.find_by_code(code, options={})
+      load_from_response send_request("pre-approvals/#{code}", options[:credentials])
+    end
+
+    def self.search_by_days_interval(options={})
+      # Default options
+      options = {
+        interval: 30,
+        per_page: 50,
+        page: 0
+      }.merge(options)
+
+      SubscriptionSearch.new('pre-approvals/notifications', options)
+    end
+
+    def self.search_by_date_interval(options={})
+      # Default options
+      options = {
+        starts_at: Time.now,
+        ends_at: Time.now - TEN_DAYS_IN_SECONDS,
+        per_page: 50,
+        page: 0
+      }.merge(options)
+
+      SubscriptionSearch.new('pre-approvals', options)
+    end
+
+    # Serialize the HTTP response into data.
+    def self.load_from_response(request) # :nodoc:
+      subscription = new
+      Response.new(request, subscription).serialize(:search)
+
+      subscription
+    end
+
+    # Send a get request to API version, with the path given
+    def self.send_request(path, credentials)
+      Request.get_with_auth_on_url(path, API_VERSION, credentials)
+    end
+
+    def self.load_from_xml(xml)
```

Comment:
No. Because it is used in https://github.com/pagseguro/ruby/pull/205/files#diff-16812dff604a94d8d56d04621d9111dfR32.